### ISSUE TITLE
SVG fill="url(#…)" doesnt work if opacity set

### ISSUE THREAD

**Issue by mmaciola** [Samsung/thorvg#381]
Svg images that use fill="url(#..)", such as `radialgrad1.svg` and `lineargrad1.svg`, won't display if opacity is set.
Example code:
```
auto svg = tvg::Picture::gen();
if (svg->load(EXAMPLE_DIR"/lineargrad1.svg") != tvg::Result::Success) return;
svg->opacity(100);

canvas->push(move(svg));
```

**Comment by mgrudzinska** [comment]
this is solved in #326 

Because the opacity < 255 is set, the composition is applied, but currently there is no code to handle such a case (only alpha mask and inv mask are handled, since the 'if else' statements are not well handled).
In #326 the code is refactored - a function is added, that decides which secondary funct should be called, so all cases are handled properly.

**Comment by mgrudzinska** [comment]
since #326 will be splitted into smaller patches, the problem is solved in #384

**Comment by mmaciola** [comment]
After patch works ok
